src/inmanta/util/__init__.py

```diff
@@ -888,37 +887,37 @@ def remove_comment_part(to_clean: str) -> str:
     return drop_comment
 
 
+"""
+A CanonicalRequirement is a packaging.requirements.Requirement except that the name of this Requirement is canonicalized, which
+allows us to compare requirements without dealing afterwards with the format of these requirements.
+"""
 CanonicalRequirement = typing.NewType("CanonicalRequirement", packaging.requirements.Requirement)
 
 
 def parse_requirement(requirement: str) -> CanonicalRequirement:
     """
-    To be able to compare requirements, we need to make sure that every requirement's name is canonicalized otherwise issues
-    could arise when checking if packages are installed in a particular Venv.
-    This function supposes to receive an actual requirement. Commented strings will not be handled and result in a ValueError
+    Parse the given requirement string into a requirement object with a canonicalized name, meaning that we are sure that
+    every CanonicalRequirement will follow the same convention regarding the name. This will allow us compare requirements.
+    This function supposes to receive an actual requirement.
 
     :param requirement: The requirement's name
     :return: A new requirement instance
-    :raise: ValueError when commented or empty strings are provided
     """
-    # Packaging Requirement is not able to parse requirements with comment. Therefore, we need to remove the `comment` part
-    drop_comment = remove_comment_part(to_clean=requirement)
-
-    if drop_comment.startswith("#") or len(drop_comment) == 0:
-        raise ValueError(f"The requirement is invalid: Cannot be a comment or empty -> `{drop_comment}`!")
-
     # We canonicalize the name of the requirement to be able to compare requirements and check if the requirement is
     # already installed
-    # This instance is considered as doomed because the requirement name that this instance is using is not "canonicalized"
-    requirement_instance = packaging.requirements.Requirement(requirement_string=drop_comment)
+    # /!\ The following line could cause issue because we are not supposed to modify fields of an existing instance
```

Comment:
Ok, I see now. Perhaps make it a bit less alarming by expanding?
```suggestion
    # /!\ The following line could cause issue because we are not supposed to modify fields of an existing instance
    # The version of packaging is constrained to ensure this can not cause problems in production.
```

Is the "/!\" intentional?

Reply:
The "/!\" was intentional but maybe a bit too alarmist, I have removed it

Reply:
Oh, it's a danger triangle? I was very confused by the slashes.